components-registry-service-server/src/main/kotlin/org/octopusden/octopus/components/registry/server/service/impl/ComponentRegistryResolverImpl.kt

```
@@ -185,8 +199,71 @@ class ComponentRegistryResolverImpl(
         return result
     }
 
+    override fun findComponentsByDockerImages(images: Set<Image>): Set<ComponentImage> {
+        val imageNames = images.map { it.name }.toSet()
+        return buildImageToComponentMap().filterKeys(imageNames::contains).mapNotNull { (imgName, component) ->
+            images.find { it.name == imgName }?.let { requiredImage ->
+                findConfigurationByImage(imgName, requiredImage.tag, component)
+            }
+        }.toSet()
+    }
+
+    private fun findConfigurationByImage(imageName: String, imageTag: String, compId: String): ComponentImage? {
+
+        var (versionString, tagSuffix) = extractVersionAndSuffix(imageTag)
+
+        try {
+            versionString = getJiraComponentVersion(compId, versionString).version
+        } catch (e: NotFoundException) {
+            LOG.debug("Component $compId with version $versionString not found, possible no Jira section defined. It's ok for test data")
```

Is it really possible case? Version formats are to be mandatory properties.

---

we have some test data with no version like test-common/src/test/resources/components-registry/common/TestComponents.groovy

---

yes it does

---

We get this Exception if try to resolve by the version that is not matched to any version range in the component.
For example when resolving <myImage, 3.0>
```
myComp {
  "(1.0, 2.0)" {
    distribution {
      docker = "myImage"
    }
  }
}
```

And in this case we ignore this error. I don't think it's good that we hide this problem. At least it should be warn.
And the text is incorrect